extensions/sql_table/js/sql_table.html

```diff
@@ -0,0 +1,246 @@
+
+             thead.appendChild(headerRow);
+             table.appendChild(thead);
+         }
+
+         // Create data rows
+         const tbody = document.createElement('tbody');
+         results.forEach(row => {
+             const tr = document.createElement('tr');
+             Object.values(row).forEach(value => {
+                 const td = document.createElement('td');
+                 td.innerHTML = value;
+                 tr.appendChild(td);
+             });
+             tbody.appendChild(tr);
+         });
+
+         table.appendChild(tbody);
+         return table;
+     }
+
+
+     try {
+         // Initialize SQL.js
+         const SQL = await initSqlJs({
+             locateFile: file => `https://cdnjs.cloudflare.com/ajax/libs/sql.js/1.8.0/${file}`
+         });
+
+         // Create a database
+         const db = new SQL.Database();
+
+         // Convert table to JSON
+         const data = tableToJson(tableElement);
+
+         // Create table and insert data
+         if (data.length > 0) {
+             // Generate CREATE TABLE statement
+             const columns = Object.keys(data[0])
+                                   .map(col => `"${col}" TEXT`)
+                                   .join(', ');
+
+             db.run(`CREATE TABLE ${tableName} (${columns})`);
+
+             // Insert data
+             data.forEach(row => {
+                 const columns = Object.keys(row).map(col => `"${col}"`).join(', ');
+                 const values = Object.values(row).map(val => `'${val}'`).join(', ');
+                 db.run(`INSERT INTO ${tableName} (${columns}) VALUES (${values})`);
+             });
+
+             // Execute query
+             const results = db.exec(sqlQuery);
+
+             // Convert results to array of objects
+             if (results.length > 0) {
+                 const columns = results[0].columns;
+                 const rows = results[0].values.map(row => {
+                     return columns.reduce((obj, col, i) => {
+                         obj[col] = row[i];
+                         return obj;
+                     }, {});
+                 });
+
+                 // Convert to HTML table and return
+                 return createResultTable(rows);
+             }
+         }
+
+         // Return empty table if no results
+         return createResultTable([]);
+
+     } catch (error) {
+         console.error('Error executing query:', error);
+         const errorTable = document.createElement('table');
+         // Copy class attribute from input table
+         if (tableElement.hasAttribute('class')) {
+             errorTable.setAttribute('class', tableElement.getAttribute('class'));
+         }
+         errorTable.innerHTML = `<tr><td>Error executing query: ${error.message}</td></tr>`;
+         return errorTable;
+     }
+ }
+
+ function initializeTableQueries() {
+     // Find all tables in the document
+     const tables = document.getElementsByTagName('table');
```

Comment:
defenitly we need a flag to turn on/off this feature, maybe a `data-*` attribute like `data-queryable` but I'm not sure how to support this from Markdown without writing raw HTML table.

Reply:
yeah, some marker on the table, we don't even need anything verbose,

Reply:
another way to do that is not added this field+button on the top of every table but have it as a shortcode that renders it.  like if you have a line `/sql-table` in a single line it renders the form (shortcodes are already a feature in xlog) .

in this case querying table in a page is a choice you have to make as a user not a passive feature. so it requires an effort. 
and it doesn't solve the problem if you want to turn it on/off per table. it only solve it if you want to turn it on/off per page. 

Another way is to have this form popup only if you hover on the table. so the UI will be less cluttered